### PR TITLE
feat: Setups prisma accelerate

### PR DESCRIPTION
**Pull request by zomars**
# What does this PR do?

- Enables Prisma Accelerate extension
- Suppress some resulting Vitest errors until we figure out the new typings (test are passing tho)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/test/lib/checkBookingLimits.test.ts [hunk 1]
```diff
@@ -23,20 +23,23 @@ const MOCK_DATA: Mockdata = {
 
 describe("Check Booking Limits Tests", () => {
   it("Should return no errors", async () => {
+    // @ts-expect-error Prisma v5 typings are not yet available
```
Comment (zomars) [inline apps/web/test/lib/checkBookingLimits.test.ts:26]: Prisma client with extensions break Vitest deep Mocking

File: packages/features/auth/lib/getServerSession.ts [hunk 2]
```diff
@@ -50,6 +50,8 @@ export async function getServerSession(options: {
     where: {
       email: token.email.toLowerCase(),
     },
+    // TODO: Re-enable once we get confirmation from compliance that this is okay.
+    // cacheStrategy: { ttl: 60, swr: 1 },
```
Comment (zomars) [inline packages/features/auth/lib/getServerSession.ts:54]: Until we get confirmation from compliance @keithwillcode @ConnorGargano

**Review by zomars — COMMENTED** [review]
Ready to review

File: apps/web/test/lib/checkBookingLimits.test.ts (hunk 1 above)
Comment (leog) [reply apps/web/test/lib/checkBookingLimits.test.ts:26]: Is there any estimate on when this is coming? Maybe it is worth creating a follow-up issue to tackle this when the time comes

**Review by leog — DISMISSED** [review]
LGTM

File: packages/features/ee/teams/lib/getUserAdminTeams.ts [hunk 3]
```diff
@@ -82,7 +82,8 @@ const getUserAdminTeams = async ({
         name: user.name || "me",
         logo: user?.avatar === "" ? null : user?.avatar,
         isUser: true,
-        ...(includeCredentials && { credentials: user.credentials }),
+        credentials: includeCredentials ? [] : user.credentials,
+        parent: null,
```
Comment (emrysal) [inline packages/features/ee/teams/lib/getUserAdminTeams.ts:86]: is this an unrelated bug?

**Comment by zomars** [comment]
Pushed public API build fixes. Should be good to go.

**Review by PeerRich — APPROVED** [review]

**Pull request merged**